mmengine/runner/deepspeed_runner.py

```diff
@@ -0,0 +1,771 @@
+            self.cfg.get('model_wrapper_cfg'), self.model)
+
+        # build optim wrapper
+        self.optim_wrapper = self.build_optim_wrapper(self.optim_wrapper)
+        assert isinstance(self.optim_wrapper, DeepSpeedOptimWrapper), (
+            'OptimWrapper type should be \'DeepSpeedOptimWrapper\' when using '
+            'DeepSpeedRunner.')
+
+        # initialize DeepSpeed Engine
+        self.model, optimizer, _, _ = deepspeed.initialize(
+            model=self.model,
+            optimizer=self.optim_wrapper.optimizer,
+            model_parameters=self.model.parameters(),
+            config=self.ds_config)
+        self.optim_wrapper.optimizer = optimizer
+
+        # Set logging level to remove duplicate training log from DeepSpeed
+        deepspeed_logger = logging.getLogger('DeepSpeed')
+        deepspeed_logger.setLevel(logging.WARNING)
+
+        self.inject_basemodel_methods()
+
+        # get model name from the model class
+        if hasattr(self.model, 'module'):
+            self._model_name = self.model.module.__class__.__name__
+        else:
+            self._model_name = self.model.__class__.__name__
+
+        self._hooks: List[Hook] = []
+        # register hooks to `self._hooks`
+        self.register_hooks(default_hooks, custom_hooks)
+
+        # dump `cfg` to `work_dir`
+        self.dump_config()
+
+    def wrap_model(self, model_wrapper_cfg: Optional[Dict],
+                   model: BaseModel) -> Union[BaseModel, PipelineModule]:
+        """Wrap the model to :obj:``PipelineModule`` or other custom
+        distributed data-parallel module wrappers.
+
+        An example of ``model_wrapper_cfg``::
+
+            model_wrapper_cfg = dict(
+                type='PipelineModule',
+            )
+
+        Args:
+            model_wrapper_cfg (dict, optional): Config to wrap model. If not
+                specified, ``BaseModel`` will be used. Defaults to None.
+            model (BaseModel): Model to be wrapped.
+
+        Returns:
+            BaseModel or PipelineModule: BaseModel or ``PipelineModule``.
+        """
+        if is_model_wrapper(model):
+            if model_wrapper_cfg is not None:
+                raise TypeError(
+                    'model has been wrapped and "model_wrapper_cfg" should be '
+                    f'None, but got {model_wrapper_cfg}')
+
+            return model
+
+        # Set `export CUDA_VISIBLE_DEVICES=-1` to enable CPU training.
+        model = model.to(get_device())
+
+        if not self.distributed:
+            self.logger.info(
+                'Distributed training is not used, all SyncBatchNorm (SyncBN) '
+                'layers in the model will be automatically reverted to '
+                'BatchNormXd layers if they are used.')
+            model = revert_sync_batchnorm(model)
+            return model
+
+        if model_wrapper_cfg is None:
+            # Model will be wrapped in `deepspeed.initialize`.
+            pass
```

Comment:
Wrapping model with `DeepSpeedEngine` is done in `deepspeed.initialize`.
We can get ouf its logic by wrapping `DeepSpeedEngine` as `model_wrapper`.